test/e2e/providers/aws/aws.go

```diff
@@ -325,14 +324,16 @@ func (a *awsProvider) GenerateMachineSet(withWindowsLabel bool, replicas int32)
 	if err != nil {
 		return nil, fmt.Errorf("unable to get subnet: %v", err)
 	}
-	machineSetName := "e2e-windows-machineset-"
+	machineSetName := "e2e"
 	matchLabels := map[string]string{
 		"machine.openshift.io/cluster-api-cluster": clusterName,
 	}
 
 	if withWindowsLabel {
-		matchLabels[windowsLabel] = "Windows"
-		machineSetName = machineSetName + "with-windows-label-"
+		matchLabels[clusterinfo.MachineOSIDLabel] = "Windows"
+		// Designate MachineSets that set a Windows label on the Machine with
+		// "e2e-wm", to signify they should be configured by the Windows Machine controller
+		machineSetName += "-wm"
```

Comment:
So you have copy pasted this code to three places and will have to do so as add more providers. Isn't this a hint that it needs to be modularized?